test/test-framework-spec.md

```diff
@@ -0,0 +1,860 @@
+  <tr>
+   <td>@test:assertNotEquals
+   </td>
+   <td>Checks if the specified value is not equal to the expected value
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertTrue
+   </td>
+   <td>Checks if the specified value is true
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertFalse
+   </td>
+   <td>Checks if the specified value is false
+   </td>
+  </tr>
+  <tr>
+   <td>@test:assertFail
+   </td>
+   <td>Forces a test case to fail
+   </td>
+  </tr>
+</table>
+
+
+Each assertion allows providing an optional assertion fail message. 
+
+**Example**
+
+
+```ballerina
+@test:Config {}
+function testAssertIntEquals() {
+    int answer = 0;
+    int a = 5;
+    int b = 3;
+    answer = intAdd(a, b);
+    test:assertEquals(answer, 8, msg = "int values not equal");
+}
+```
+
+
+### Grouping Tests
+
+Organizing tests into groups allows the user to run a specific group of tests using the `--groups` option. 
+
+**Example**
+
+```ballerina
+@test:Config { groups: ['group1'] }
+function test1() {
+    // Test code ...
+}
+```
+
+### Enabling / Disabling Tests
+
+Individual test cases can be disabled by adding the `enable : false` option in the test config annotation. This makes sure that the particular test case is skipped when running the tests. 
+
+```ballerina
+@test:Config { enable : false }
+function test1() {
+    // Test code ...
+}
+```
+
+When a test case is disabled, before and after functions specified in the test configurations and all the dependent
+ test cases will be skipped. 
+
+
+### Order of Execution
+
+The depends on attribute allows the user to define a list of function names that the test function depends on. These
+functions will be executed before the test execution. The order in which the comma separated list of functions appears
+has no prominence and thus will be executed in an arbitrary manner. This attribute can be used to ensure that the
+tests are being executed in the expected order.
+
+
+```ballerina
+@test:Config { }
+function test1() {
+    // Test code ...
+}
+
+@test:Config { dependsOn: ["test1"] }
+function test2() {
+    // Test code ...
+}
+```
+
+## Mocking
+
+The mocking support in the Ballerina test framework provides capabilities to mock a function or an object for unit testing. 
+The mocking feature can be used to control the behavior of functions and objects by defining return values or
+ replacing the entire object/function with a user-defined equivalent. This feature will help the user to test the
+  Ballerina code independently from other modules and external endpoints.
+
+
+### Mock Annotation
+
+Initializing a function mock needs a preceding annotation in order to identify and replace the occurrence of the original function during compilation.
+
+This annotation is only required when mocking functions since a part of function mocking is handled in compile time. Mocking an object is completely handled in the runtime and therefore, this annotation is not required when initializing a mock for an object.
```

Comment:
```suggestion
This annotation is only required when mocking functions since a part of function mocking is handled during the compile time. Mocking an object is completely handled in the runtime, and thereby, this annotation is not required when initializing a mock for an object.
```